iceoryx_utils/include/iceoryx_utils/internal/cxx/string.inl

```diff
@@ -265,6 +301,34 @@ inline string<Capacity>::operator std::string() const noexcept
     return std::string(c_str());
 }
 
+template <uint64_t Capacity>
+template <uint64_t N>
+inline string<Capacity>& string<Capacity>::copy(const string<N>& rhs) noexcept
+{
+    static_assert(N <= Capacity,
+                  "Assignment failed. The capacity of the given fixed string is larger than the capacity of this.");
+    uint64_t strSize = rhs.size();
+    std::memcpy(m_rawstring, rhs.c_str(), strSize);
+    m_rawstring[strSize] = '\0';
+    m_rawstringSize = strSize;
+    return *this;
+}
+
+template <uint64_t Capacity>
+template <uint64_t N>
+inline string<Capacity>& string<Capacity>::move(string<N>&& rhs) noexcept
+{
+    static_assert(N <= Capacity,
```

Comment:
I would perform here the self move check
```
if ( this == &rhs ) {
...
```

Then you can remove the self move check in the move assignment operator and the move method is safe to use.

Reply:
To avoid code duplication I call `move` also in the move constructors where this check is unnecessary.